js/ui/form/ui.form.js

```diff
@@ -829,7 +829,11 @@ var Form = Widget.inherit({
         this._renderLayout();
         this._renderValidationSummary();
 
-        this._cachedScreenFactor = this._getCurrentScreenFactor();
+        if(this._targetScreenFactor) {
+            this._lastMarkupScreenFactor = this._targetScreenFactor;
+        } else {
+            this._lastMarkupScreenFactor = this._getCurrentScreenFactor();
+        }
```

Comment:
Maybe make simpler: `this._lastMarkupScreenFactor = this._targetScreenFactor || this._getCurrentScreenFactor()`